snapcraft/internal/sources/__init__.py

```diff
@@ -164,7 +165,14 @@ def get_source_handler(source, *, source_type=''):
     if not source_type:
         source_type = _get_source_type_from_uri(source)
 
-    return _source_handler.get(source_type, Local)
+    handler = _source_handler.get(source_type, None)
+
+    if not handler:
+        raise ValueError((
+            'invalid snapcraft.yaml: '
+            'unknown source type: {}').format(source_type))
```

Comment:
@sergiusens can you please check if this error is consistent with our other messages?
I think it could be something like:
ValueError('no handler to manage source type ({})'.format(source_type))